### ISSUE TITLE
password reset  mails are not coming to the mail box

### ISSUE THREAD

**Issue by arun767** [ltb-project/self-service-password#498]
Hi,

I have build in my lab with rpm7 package everything was working as expected and now i am building with rpm 8 package in centos 8. i have only enabled three option, with old password, with question and with mail. i have copied and replaced my config file in the new server  and except mail other two are working fine.

when i select email option, i have received a single mail with the  password reset link and when i tried to send again , mails are not getting delivered.  i have tried multiple times with different time intervals  but mails to getting delivered. Able to telnet to the smtp server there is no logs also..

**Comment by arun767** [comment]
I did lot of troubleshooting with limited knowledges and unable to find why mail are not getting delivered

$mail_sendmailpath = '/usr/sbin/sendmail';
$mail_protocol = 'smtp';
$mail_smtp_debug = 0 ;
$mail_debug_format = 'error_log';
$mail_smtp_host = 'xxxx';
$mail_smtp_auth = false;
$mail_smtp_user = '';
$mail_smtp_pass = '';
$mail_smtp_port = 587;
$mail_smtp_timeout = 30;
$mail_smtp_keepalive = false;
$mail_smtp_secure = '';
$mail_smtp_autotls = false;
$mail_contenttype = 'text/plain';
$mail_wordwrap = 0;
$mail_charset = 'utf-8';
$mail_priority = 3;
$mail_newline = 'PHP_EOL';
$mail_smtp_options = array();
$obscure_failure_messages = array("mailnomatch");

i don't know what wrong i have done. is there any prerequisites for mail.


SERVER -> CLIENT: 220 server Microsoft ESMTP MAIL Service, Version: 10.0.17763.1697 ready at Sun, 18 Apr 2021 19:16:13 +0530
CLIENT -> SERVER: EHLO prp.XXXcom
SERVER -> CLIENT: 250-server Hello [172.21.122.87]250-TURN250-SIZE 2097152250-ETRN250-PIPELINING250-DSN250-ENHANCEDSTATUSCODES250-8bitmime250-BINARYMIME250-CHUNKING250-VRFY250-TLS250-STARTTLS250 OK
CLIENT -> SERVER: MAIL FROM:<prp@xxx.com>
SERVER -> CLIENT: 250 2.1.0 prp@xxx.com....Sender OK
CLIENT -> SERVER: RCPT TO:<arun.@xxxx.com>
SERVER -> CLIENT: 250 2.1.5 arun@xxx.com
CLIENT -> SERVER: DATA
SERVER -> CLIENT: 354 Start mail input; end with <CRLF>.<CRLF>
CLIENT -> SERVER: Date: Sun, 18 Apr 2021 13:46:13 +0000PHP_EOLTo: arun@xxx.comPHP_EOLFrom: Self Service Password Administrator <prp@xxx.com>PHP_EOLReply-To: Self Service Password Administrator <prp@xxx.com>PHP_EOLSubject: Reset your passwordPHP_EOLMessage-ID: <07238962ccbe1115623a15fd2c4a5be8@prp.xxx.com>PHP_EOLX-Priority: 3PHP_EOLX-Mailer: PHPMailer 5.2.23 (https://github.com/PHPMailer/PHPMailer)PHP_EOLMIME-Version: 1.0PHP_EOLContent-Type: text/plain; charset=utf-8PHP_EOLPHP_EOLHello arun,PHP_EOLPHP_EOLClick here to reset your password:PHP_EOLhttps://prp.xxx.com/index.php?action=resetbytoken&token=3vUCAOYRTb5NJALBSkZSTyfOHGCL%2FXQ86kzsxTqwO70XJF2fj%2BjWle6e%2Fzh6SKfRSnJlcnziSgpuTJ%2FIHrgDEBCgfT%2FYj3ld3C7sTTmijNRzRBICt93XYBo01kRrXQXaqXjYtD0j2kjW3DOFdg8%3DPHP_EOLPHP_EOLIf you didn't request a password reset, please ignore this email.Self Service PasswordPHP_EOL
CLIENT -> SERVER: .
SERVER -> CLIENT: 250 2.6.0 <ednH00005cfb@> Queued mail for delivery
CLIENT -> SERVER: QUIT
SERVER -> CLIENT: 221 2.0.0  Service closing transmission channel

**Comment by coudot** [comment]
Hello,

first noticeable issue:
```php
$mail_newline = 'PHP_EOL';
```

It should be:
```php
$mail_newline = PHP_EOL;
```

Then, using SSP 1.3 on RHEL8 may not work. We are working on 1.4 version, that will come with an updated version of PHPMailer. You should try it.

**Comment by arun767** [comment]
Thanks, I have removed the quote after that when i click on send mail i got a mail. but when when i click again, no mail.

**Comment by coudot** [comment]
Try to compare logs between the working case and the failing case.

**Comment by arun767** [comment]
i just removed the single quote only.  is there compactable issue with PHP version. currently i am running with PhP ver 7.4

**Comment by faust64** [comment]
The logs your showing are pretty normal, beyond the PHP_EOL stuff.

As you can see, the message was already being queued in your previous test (confirmed by the server, replying with `Queued mail for delivery`).
If you still are missing messages, you should investigate on that `Microsoft ESMTP MAIL Service` relay. That's the one with your message in queue.

**Comment by arun767** [comment]
yes i saw that. i checked in the smtp server and no connection and queue in there  . also i have similar test setup where everything is working fine. the only difference is in my test  server php version is 7.3 and in prod  ver 7.4..

**Comment by faust64** [comment]
Your log mentions `SERVER -> CLIENT: 250 2.6.0 <ednH00005cfb@> Queued mail for delivery`. Thus, there should be logs in that remote relay, that would tell us what went on with that message ID (`ednH00005cfb@`).
Given the logs so far you've shown so far, we can't tell you where is went, though for sure it left your server and was accepted by some third-party SMTP relay.

Having fixed your configuration, look for the message ID of your last test message, and check that Microsoft mail relay for corresponding events. In doubt, maybe check the recipient junk folder?

**Comment by arun767** [comment]
I have checked in the SMTP server logs and there is no  error logs,

Notice a strange thing. When i call the server in the browser via IP address like https://192.168.1.1 , i am getting mails  but when i use server name like https://prp.abc.com, i am not getting mails.  strange  but i have checked  verified multiple times.
